### PR TITLE
optim/lr_scheduler: add WarmupCosineDecayLR scheduler

### PR DESCRIPTION
**Pull request by GitGlimpse895**
Summary
Add WarmupCosineDecayLR, a first-class linear-warmup + cosine-decay learning rate scheduler to torch.optim.lr_scheduler. This is the dominant LR schedule used in modern deep learning (Transformers, ViTs, LLMs like GPT/BERT/LLaMA) but currently has no native PyTorch implementation. Users are forced to compose SequentialLR(LinearLR, CosineAnnealingLR) manually, which has a correctness bug: because SequentialLR never calls _get_closed_form_lr on its sub-schedulers, resuming training from a checkpoint with a non-default last_epoch silently restores the wrong learning rate.
Changes

torch/optim/lr_scheduler.py: Add WarmupCosineDecayLR class with correct get_lr and _get_closed_form_lr implementations. Add to __all__.
torch/optim/__init__.py: Export WarmupCosineDecayLR at the torch.optim top level. Add to __all__.

Usage
scheduler = torch.optim.WarmupCosineDecayLR(optimizer, T_max=10_000, warmup_steps=500, eta_min=1e-6)
Test Plan
Verified linear warmup produces correct LR values at step 0, mid-warmup, and at warmup_steps. Verified cosine decay produces correct LR values immediately after warmup and at T_max. Verified _get_closed_form_lr returns bit-exact values matching get_lr at every step, ensuring correct checkpoint resume behaviour. Verified ValueError is raised for invalid T_max and warmup_steps arguments.

### CODE REVIEW

**Comment by janeyx99** [comment]
Hi, thanks for opening this PR. @diputs-sudo is looking into LRScheduler inadequacies and the team is considering revamping LRScheduler in general, so I'm not inclined to land anything new into LRScheduler at this moment. I'd rather opt for fixes. That said, this is a good consideration for a redesign cc @diputs-sudo
